src/App.tsx

```diff
@@ -1,12 +1,19 @@
-import React from 'react';
+import React, { useContext } from 'react';
 
 import Home from './components/Home/Home';
 import MainLayout from './components/Layout/MainLayout';
+import { UserContext } from './contexts/UserContext';
 
-const App: React.FC = () => (
-  <MainLayout>
-    <Home />
-  </MainLayout>
-);
+const App: React.FC = () => {
+  const { user, setUser } = useContext(UserContext);
+
+  user !== null && console.log('user is authenticated');
```

Comment:
Move this into useEffect

Reply:
the `useEffect` was defined in the `UserProvider` to invoke the `getUser()` each time when `UserProvider` mounts.
But as requested I changed it to invoke the `getUser()` each time the `App` mounts. 
and the `getUser()` has also been sent to the `App` via Context. Could you let me know which way is correct/preferred?